low-level/sensors/impl/centos_7/service_monitor.py

```diff
@@ -235,15 +240,22 @@ def properties_changed_handler(self, interface, changed_properties,
                     Service.active_services.add(self.name)
                 if self._service_state is InactiveState:
                     self.new_service_state(ActiveState)
+                if self.name in Service.failed_services:
+                    Service.failed_services.discard(self.name)
```

Comment:
this new_service_state() should take care of discarding service name from other state lists, or its becoming too much repetitive code
If not seen straightforward to address, keep a TODO